skbio/metadata/_testing.py

```diff
@@ -904,3 +905,166 @@ def test_has_positional_metadata(self):
         obj = self._positional_metadata_constructor_(
             2, positional_metadata={'foo': [1, 2], 'bar': ['abc', 'def']})
         self.assertTrue(obj.has_positional_metadata())
+
+
+class IntervalMetadataMixinTests:
+    def _set_up(self):
+        self.upper_bound = 9
+        self.im = IntervalMetadata(self.upper_bound)
+        self.intvls = [
+            {'locations': [(0, 1)], 'metadata': {'gene': 'sagA'}},
+            # repeat it
+            {'locations': [(0, 1)], 'metadata': {'gene': ['a'],
+                                                 'product': 'foo'}}]
+
```

Comment:
Would it be worthwhile to have a different interval other than (0, 1) or multiple intervals? It would be nice to double check if the equality works as anticipated.